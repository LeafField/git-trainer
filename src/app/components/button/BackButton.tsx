"use client";
import React from "react";
import Link from "next/link";
import Image from "next/image";
import backImage from "../../../../public/Arrow.png";

const BackButton = () => {
  return (
    <Link
      className="flex h-[2.8125rem] w-[8.5625rem] items-center justify-center border border-white"
      href="/"
    >
      <figure className="relative h-[23px] w-[74px] transition-all hover:translate-x-[-10px] hover:scale-x-[1.1]">
        <Image
          src={backImage}
          alt="トップページに戻るボタン"
          style={{ objectFit: "cover", transition: "all 0.25s ease-in-out" }}
          fill
          sizes="72px"
          priority
          placeholder="blur"
          className="transition-all"
        />
      </figure>
    </Link>
  );
};

export default BackButton;
