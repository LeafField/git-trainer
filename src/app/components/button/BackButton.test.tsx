import { render, screen } from "@testing-library/react";
import { composeStories } from "@storybook/react";
import * as stories from "./BackButton.stories";

const { BackButtonTemplate } = composeStories(stories);

test("BackButtonがきちんとレンダリングされているか", () => {
  const { container } = render(<BackButtonTemplate />);
  expect(screen.getByRole("link")).toBeInTheDocument();
  expect(screen.getByRole("img")).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
