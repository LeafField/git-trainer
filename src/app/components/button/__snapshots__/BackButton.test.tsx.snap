// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButtonがきちんとレンダリングされているか 1`] = `
<div>
  <a
    class="flex h-[2.8125rem] w-[8.5625rem] items-center justify-center border border-white"
    href="/"
  >
    <figure
      class="relative h-[23px] w-[74px] transition-all hover:translate-x-[-10px] hover:scale-x-[1.1]"
    >
      <img
        alt="トップページに戻るボタン"
        class="transition-all"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="72px"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 16w, /_next/image?url=%2Fimg.jpg&w=32&q=75 32w, /_next/image?url=%2Fimg.jpg&w=48&q=75 48w, /_next/image?url=%2Fimg.jpg&w=64&q=75 64w, /_next/image?url=%2Fimg.jpg&w=96&q=75 96w, /_next/image?url=%2Fimg.jpg&w=128&q=75 128w, /_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; transition: all 0.25s ease-in-out; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
      />
    </figure>
  </a>
</div>
`;
